instrumentation/jsp-2.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jsp/HttpJspPageInstrumentationSingletons.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jsp;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.instrumentation.api.config.Config;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.jsp.HttpJspPage;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.LoggerFactory;
+
+public class HttpJspPageInstrumentationSingletons {
+  private static final boolean CAPTURE_EXPERIMENTAL_SPAN_ATTRIBUTES =
+      Config.get().getBoolean("otel.instrumentation.jsp.experimental-span-attributes", false);
+
+  private static final Instrumenter<HttpServletRequest, Void> INSTRUMENTER;
+
+  static {
+    INSTRUMENTER =
+        Instrumenter.<HttpServletRequest, Void>newBuilder(
+                GlobalOpenTelemetry.get(),
+                "io.opentelemetry.jsp-2.3",
+                HttpJspPageInstrumentationSingletons::spanNameOnRender)
+            .addAttributesExtractor(new RenderAttributesExtractor())
+            .newInstrumenter(SpanKindExtractor.alwaysInternal());
+  }
+
+  private static String spanNameOnRender(HttpServletRequest req) {
+    // get the JSP file name being rendered in an include action
+    Object includeServletPath = req.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
+    String spanName = req.getServletPath();
+    if (includeServletPath instanceof String) {
+      spanName = includeServletPath.toString();
+    }
+    return "Render " + spanName;
+  }
+
+  public static Instrumenter<HttpServletRequest, Void> instrumenter() {
+    return INSTRUMENTER;
+  }
+
+  private HttpJspPageInstrumentationSingletons() {}
+
+  private static class RenderAttributesExtractor
+      extends AttributesExtractor<HttpServletRequest, Void> {
+
+    @Override
+    protected void onStart(AttributesBuilder attributes, HttpServletRequest request) {
+      if (!CAPTURE_EXPERIMENTAL_SPAN_ATTRIBUTES) {
```

Comment:
Nit: you can just not add the extractor in the instrumenter building phase instead

Reply:
I cannot decide which way is better. Adding extractors conditionally breaks the fluent interface of `InstrumenterBuilder`